### PR TITLE
Edited the YML file to make and build the code and tests and Added trie_search with tests

### PR DESCRIPTION
**Pull request by DustinoZhang**
Added trie_search (Valerie) with tests (Valerie). Also edited YML file to make and build the library and tests (Dustin).

### CODE REVIEW

**Comment by mpcsh** [comment]
@DustinoZhang looks like the changes in this PR are more than just the YML. Please update the title & description to reflect that so that the reviewers know what they're getting into ahead of time.

**Comment by DustinoZhang** [comment]
Updated!^^

**Comment by mpcsh** [comment]
@dengyijia we need your review ASAP. Dustin has been waiting for 5 days at this point.

File: src/trie.c [hunk 1]
```diff
@@ -84,3 +81,29 @@ int insert_string(char *word, trie_t *t)
...
+    len = strlen(word);
+    curr = t;
+    next = t->children;
+
+    for (int i=0; i<len; i++) {
+        int j = (int) word[i];
+        curr = next[j];
+
+        if (curr == NULL)
+            return 0;
```
Comment (mpcsh) [inline src/trie.c]: Same issue about magic numbers.

File: src/trie.c [hunk 2]
```diff
@@ -84,3 +81,29 @@ int insert_string(char *word, trie_t *t)
...
+
+        next = next[j]->children;
+    }
+
+    if (curr->is_word == 1) 
+        return 1;
```
Comment (mpcsh) [inline src/trie.c]: Same issue about magic numbers.

File: src/trie.c [hunk 3]
```diff
@@ -84,3 +81,29 @@ int insert_string(char *word, trie_t *t)
...
+
+    return -1;
```
Comment (mpcsh) [inline src/trie.c]: Same issue about magic numbers.

File: tests/test_trie.c [hunk 4]
```diff
@@ -84,4 +84,48 @@ Test(trie, free)
...
+
+    found = trie_search("CS22000", t); // Checks extension of inserted word
+    cr_assert_eq(found, 0, "trie_search() returns %d instead of 0"
+           "  with extension of inserted word", found);
+
+    found = trie_search("Computer", t); // Checks different word
+    cr_assert_eq(found, 0, "trie_search() returns %d instead of 0"
+          "  with word different from inserted word", found);
+}
+
```
Comment (mpcsh) [inline tests/test_trie.c]: We need more tests, especially for search. How about search tests on non-singleton tries? Search tests that succeed and search tests that fail? There's a lot of your source code that's not covered by tests.

**Pull request merged**